components/site/src/link_checking.rs

```diff
@@ -101,53 +104,94 @@ pub fn check_external_links(site: &Site) -> Result<()> {
         .values()
         .map(|p| {
             let path = &p.file.path;
-            p.external_links.iter().map(move |l| (path.clone(), l))
+            p.external_links.iter().map(move |l| {
+                let domain = match Url::parse(l) {
+                    Ok(url) => match url.host_str().map(String::from) {
+                        Some(domain_str) => domain_str,
+                        None => "".to_string(),
+                    },
+                    Err(why) => panic!("{:?}", why),
+                };
+                (path.clone(), l.to_string(), domain)
+            })
         })
         .flatten();
     let section_links = library
         .sections()
         .values()
         .map(|p| {
             let path = &p.file.path;
-            p.external_links.iter().map(move |l| (path.clone(), l))
+            p.external_links.iter().map(move |l| {
+                let domain = match Url::parse(l) {
+                    Ok(url) => match url.host_str().map(String::from) {
+                        Some(domain_str) => domain_str,
+                        None => "".to_string(),
+                    },
+                    Err(why) => panic!("{:?}", why),
+                };
+                (path.clone(), l.to_string(), domain)
+            })
```

Comment:
This is identical to the `page_links` snippet above. Is that a problem? Should we put these lines a function?

Reply:
Yep, I want to add some tests for it too

Reply:
Moved domain parsing in `get_link_domain()` in https://github.com/getzola/zola/pull/1421/commits/72f21d7b578422abfa84ed6b1e912bb03fc2f430.

- Is the name ok?
- Not sure where to put it 😄 
- And also as said above, not sure about the handling of `Option` and `Result`.